### ISSUE TITLE
_id not used for upsert when part of $and-filter

### ISSUE THREAD

**Issue by vandmo** [bwaldvogel/mongo-java-server#208]
The following code illustrates the issue.

```java
            // This works as expected
            // Creates: {"_id": "as_expected", "value": 100}
            collection
                    .findOneAndUpdate(
                            BsonDocument.parse("{\"_id\": \"as_expected\"}"),
                            BsonDocument.parse("{\"$set\": {\"value\": 100}}"),
                            new FindOneAndUpdateOptions().upsert(true));

            // This generates an identifier instead of using the one in the $and-list
            // Creates: {"_id": {"$oid": "f6a40f2c1855632c12f51c59"}, "value": 200}
            collection
                    .findOneAndUpdate(
                            BsonDocument.parse("{\"$and\": [{\"_id\": \"this_goes_gone\"}]}"),
                            BsonDocument.parse("{\"$set\": {\"value\": 200}}"),
                            new FindOneAndUpdateOptions().upsert(true));
```

The same request towards the official MonoDB would find and use the _id supplied in the query/filter.

**Comment by vandmo** [comment]
Full example:
```java
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.model.FindOneAndUpdateOptions;
import de.bwaldvogel.mongo.MongoServer;
import de.bwaldvogel.mongo.backend.memory.MemoryBackend;
import org.bson.BsonDocument;
import org.bson.Document;

class ReproduceIssue {

    public static void main(String[] args) {
        MongoServer server = new MongoServer(new MemoryBackend());
        String connectionString = server.bindAndGetConnectionString();
        try (MongoClient client = MongoClients.create(connectionString)) {
            MongoCollection<Document> collection = client
                    .getDatabase("testdb")
                    .getCollection("testcollection");

            // This works as expected
            // Creates: {"_id": "as_expected", "value": 100}
            collection
                    .findOneAndUpdate(
                            BsonDocument.parse("{\"_id\": \"as_expected\"}"),
                            BsonDocument.parse("{\"$set\": {\"value\": 100}}"),
                            new FindOneAndUpdateOptions().upsert(true));

            // This generates an identifier instead of using the one in the $and-list
            // Creates: {"_id": {"$oid": "f6a40f2c1855632c12f51c59"}, "value": 200}
            collection
                    .findOneAndUpdate(
                            BsonDocument.parse("{\"$and\": [{\"_id\": \"this_goes_gone\"}]}"),
                            BsonDocument.parse("{\"$set\": {\"value\": 200}}"),
                            new FindOneAndUpdateOptions().upsert(true));

            for (Document doc : collection.find()) {
                System.out.println(doc.toJson());
            }
        } finally {
            server.shutdown();
        }
    }

}
```

**Comment by vandmo** [comment]
Oh, nice! I will try to give it a spin locally tomorrow 👍

**Comment by vandmo** [comment]
Have confirmed that it works locally with that revision :+1: 
Thanks for fixing!

**Comment by bwaldvogel** [comment]
The fix is available via `v1.44.0`
